modules/core/src/adapter/luma.ts

```diff
@@ -176,32 +176,16 @@ export class Luma {
   }
 
   /**
-   * Override `HTMLCanvasContext.getCanvas()` to always create WebGL2 contexts.
-   * Used when attaching luma to a context from an external library does not support creating WebGL2 contexts.
-   * (luma can only attach to WebGL2 contexts).
+   * Override `HTMLCanvasContext.getCanvas()` to always create WebGL2 contexts with additional WebGL1 compatibility.
+   * Useful when attaching luma to a context from an external library does not support creating WebGL2 contexts.
    */
-  enforceWebGL2(enforce: boolean = true): void {
-    const prototype = HTMLCanvasElement.prototype as any;
-    if (!enforce && prototype.originalGetContext) {
-      // Reset the original getContext function
-      prototype.getContext = prototype.originalGetContext;
-      prototype.originalGetContext = undefined;
-      return;
+  enforceWebGL2(enforce: boolean = true, adapters: Adapter[] = []): void {
```

Comment:
How does a user register an adapter? Or are the necessary adapters already registered by default?